NAS2D/Mixer/Mixer.h

```diff
@@ -19,18 +19,16 @@ namespace NAS2D
 	class Sound;
 	class Music;
 
-
 	/**
-	 * Mixer base class.
-	 *
-	 * Provides a standard Mixer interface. The base Mixer can be used
-	 * but will act as a NULL interface.
-	 */
+	 * @brief Mixer base class.
+	 * 
+	 * Provides a standard Mixer interface. The base Mixer can be used but will act as a NULL interface.
```

Comment:
Is there a way we can set a standard line wrap width, perhaps with `.editorconfig`? I get the 80 column marker in my editor. I'd be fine with increasing that to 120 characters per line, which fits quite nicely on most modern monitors.

Reply:
This was something I was concerned about. When looking at the comments via Visual Studio's Intellisense it flat out *ignores* newlines and I do not know how to add artificial ones as the `<br>` and `\n` tags are not supported (they show up as-is). This is frustrating.

Reply:
I'm not sure I really understand the response here. Why would you be trying to force newlines? Wouldn't the point here be to auto flow the text?

Reply:
For horizontal flow, sure, but for vertical flow not so much. It's primarily a readability issue. Right now, you get giant walls of text regardless of the presence of vertical whitespace in the actual comment. There are clear delimited sections for the Brief, Parameters, and raw comment, but that's it. If the text of any of them go on too long you get the aforementioned walls of text in that section.

Reply:
I feel like there's some missing detail here.

Is it perhaps flowing the two lines together? Does it correctly handle auto brief text as such? Or does it need to be manually marked up with `@brief`?

The one line that was reflowed here doesn't seem particularly long, so I'm not sure what constitutes a wall of text.

----

The other part of this is that text was reflowed by the IDE, though not all IDEs will necessarily default to the same column marker (80 columns, 120 columns, etc.). Many IDEs do support `.editorconfig` though. Would it be possible to set a `.editorconfig` value such that most common IDEs would have the same column marker to guide how text is flowed? Doesn't matter if it's either manually or automatically flowed.

Reply:
The issue is Intellisense, the code-completion subsystem for Visual Studio and Visual Studio Code. The Doxygen interpreter does not handle comments on separate lines. It runs them all together until its internal, pre-defined, and unedittable, column length is reached then it does a basic wrap to the next line.

Example of ignored newlines: 
![NoNewlines](https://user-images.githubusercontent.com/9336881/126389359-91e388fb-0acc-4c95-8f42-db0f90f7fcaf.png)

Example of section breaks:
![SectionBreaks](https://user-images.githubusercontent.com/9336881/126389461-12ce0c69-ac8d-4228-95e8-7347406e0c9b.png)


I noticed that unknown doxygen tags, like the "warning" tag in the `MixerSDL` comment header only appears in the raw comment section where it could conceivably be missed:
![NoUnknowns](https://user-images.githubusercontent.com/9336881/126389779-9272639d-4aef-450a-81ad-129a10fdf924.png)

Example of Wall'o'Text:
![WalloText](https://user-images.githubusercontent.com/9336881/126395540-d60c89ee-f81e-4b8d-8592-d2b449c15b8c.png)


Long non-Doxygen comment blocks (as multiple single-line comments `//` or a block `/*...*/`) aren't much better, they only handle one newline per column. Multiple newlines are collapsed in to one. Additionally, regardless of the comment format, if it's over ten lines, it gets truncated.

This is why I advocate type-safety, and good function/variable names over trying to document features in comments.

If you need to explain how a function works or its acceptable use, use an actual long-form API document similar to docs.microsoft or khronos.org

Reply:
Those pictures certainly help explain what's going on. For the last picture, does the original text flow match what is displayed in the raw comment section, or was it reflowed from the original source file? I see it's from a version of TinyXML2 that's not in the project, and checking online for the original source shows a file without Doxygen comments.

----

At any rate, I feel like this discussion is moving away from the original comment I was trying to make. I'm concerned about reflow noise in diffs from IDEs attempting to adjust comments according to different desired line widths. If one IDE is set to 80, and another is set to 120, and they both want to reflow comments to their own desired length, that could get annoying. It would be nice if we could have one settings files, such as `.editorconfig`, which most common IDEs will read and respect, which sets a common line width for all the IDEs currently being used on this project.

Though yes, if we don't have long comments, because the code is written clearly with good types and good names, that would of course be best. Though I suspect there will always be at least a few somewhat long comments where max line width will matter.

Reply:
> Those pictures certainly help explain what's going on. For the last picture, does the original text flow match what is displayed in the raw comment section, or was it reflowed from the original source file? I see it's from a version of TinyXML2 that's not in the project, and checking online for the original source shows a file without Doxygen comments.
> 
I didn't realize I had cut off the comment. Here's the text as-is:
```cpp

        /** Given an attribute name, Attribute() returns the value
            for the attribute of that name, or null if none
            exists. For example:

            @verbatim
            const char* value = ele->Attribute( "foo" );
            @endverbatim

            The 'value' parameter is normally null. However, if specified,
            the attribute will only be returned if the 'name' and 'value'
            match. This allow you to write code:

            @verbatim
            if ( ele->Attribute( "foo", "bar" ) ) callFooIsBar();
            @endverbatim

            rather than:
            @verbatim
            if ( ele->Attribute( "foo" ) ) {
                if ( strcmp( ele->Attribute( "foo" ), "bar" ) == 0 ) callFooIsBar();
            }
            @endverbatim
        */
```
As you can see, they have `@verbatim/@endverbatim` doxygen tags. The Doxygen Interpreter for Intellisense does not understand them so it ignores the tags and removes anything associated with them. Additionally the description section collapses any consecutive newlines into one and then truncates after 10 lines.

The Raw comment section is preserved up to 10 lines as well before being truncated.

> At any rate, I feel like this discussion is moving away from the original comment I was trying to make. I'm concerned about reflow noise in diffs from IDEs attempting to adjust comments according to different desired line widths. If one IDE is set to 80, and another is set to 120, and they both want to reflow comments to their own desired length, that could get annoying. It would be nice if we could have one settings files, such as `.editorconfig`, which most common IDEs will read and respect, which sets a common line width for all the IDEs currently being used on this project.
> 
> Though yes, if we don't have long comments, because the code is written clearly with good types and good names, that would of course be best. Though I suspect there will always be at least a few somewhat long comments where max line width will matter.

I agree. I know a specified maximum column length can be enforced with `.clang-format` or `.clang-tidy`, I do not know about `.editorconfig`. Regardless, the formatting options in `.editorconfig` were removed and the only thing retained was the commands to tell Visual Studio to use `/*` style comments over `///`.

Reply:
Looks like it preserves single newlines, but collapses multiple consecutive newlines. Anyway, not something I'm overly concerned about. Sounds like a minor limitation of one particular tool, with no real workaround, and so nothing to really do about it.

----

I'm not sure if `.clang-format` settings would be picked up by Atom. I suspect `.editorconfig` settings would be though, at least for something like the margin ruler. Of course, that brings up the question of if `clang-format` would pick up a setting from `.editorconfig`, and if you can avoid specifying such a setting in two different places, which could potentially conflict with each other.

I'm perhaps somewhat inclined to ignore this concern for now.